datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionStatsResponse.java

```diff
@@ -0,0 +1,115 @@
+/**
+ *  Copyright 2020 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.connectors.kafka;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.kafka.common.TopicPartition;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.Version;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+import org.codehaus.jackson.annotate.JsonGetter;
+import org.codehaus.jackson.annotate.JsonMethod;
+import org.codehaus.jackson.annotate.JsonSetter;
+import org.codehaus.jackson.map.DeserializationConfig;
+import org.codehaus.jackson.map.DeserializationContext;
+import org.codehaus.jackson.map.JsonDeserializer;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.map.module.SimpleModule;
+import org.codehaus.jackson.type.TypeReference;
+
+import com.linkedin.datastream.common.JsonUtils;
+
+
+/**
+ * Response structure used for Topic partition stats
+ * @see AbstractKafkaConnector#process(String)
+ */
+public class KafkaTopicPartitionStatsResponse {
+
+  private String _consumerGroupId;
+
+  private Set<TopicPartition> _topicPartitions;
+
+  /**
+   *  Default constructor for deserialization
+   */
+  public KafkaTopicPartitionStatsResponse() {
+  }
+
+  /**
+   * Constructor for KafkaTopicPartitionStatsResponse
+   * @param consumerGroupId identifier for consumer group
+   * @param topicPartitions a map of consumer offsets for topic partitions
+   */
+  public KafkaTopicPartitionStatsResponse(String consumerGroupId, Set<TopicPartition> topicPartitions) {
+    _consumerGroupId = consumerGroupId;
+    _topicPartitions = topicPartitions;
+  }
+
+  @JsonGetter("_consumerGroupId")
+  public String getConsumerGroupId() {
+    return _consumerGroupId;
+  }
+
+  @JsonSetter("_consumerGroupId")
+  public void setConsumerGroupId(String consumerGroupId) {
+    _consumerGroupId = consumerGroupId;
+  }
+
+  @JsonGetter("_topicPartitions")
+  public Set<TopicPartition> getTopicPartitions() {
+    return _topicPartitions;
+  }
+
+  @JsonSetter("_topicPartitions")
+  public void setTopicPartitions(Set<TopicPartition> topicPartitions) {
+    _topicPartitions = topicPartitions;
+  }
+
+  /**
+   * Serialize to JSON
+   */
+  public static String toJson(List<KafkaTopicPartitionStatsResponse> obj) {
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
Could you, please, check if you can reuse this object for serialization and deserialization (e.g. declare as a `static final`)? I don't think we need to instantiate and configure one every time this method is invoked. Same goes for `fromJson()`.

Reply:
I cannot make it as final. But I have made it static.